### PR TITLE
Assistants v2

### PR DESCRIPTION
**Pull request by dnnkeeper**
This is a small rough update to v2 beta helpers with support for delta streaming and new authorization using project keys. It also includes updates to AudioTranscriptionRequest to handle encoded audio streams (I'm using Vorbis encoding in my project, which I may post later as a sample).
I don't have time to create proper test cases and documentation, so you can use this branch as a reference for future updates if you find it useful.

### CODE REVIEW

File: OpenAI/Packages/com.openai.unity/Runtime/Authentication/OpenAIAuthInfo.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public sealed class OpenAIAuthInfo : IAuthInfo
         internal const string SessionKeyPrefix = "sess-";
         internal const string OrganizationPrefix = "org-";
 
-        public OpenAIAuthInfo(string apiKey, string organizationId = null)
+        public OpenAIAuthInfo(string apiKey, string organizationId = null, string project = null)
```
Comment (StephenHodgson) [inline OpenAI/Packages/com.openai.unity/Runtime/Authentication/OpenAIAuthInfo.cs:18]: does this id have a prefix?

File: OpenAI/Packages/com.openai.unity/Runtime/Audio/AudioTranscriptionRequest.cs [hunk 2]
```diff
@@ -109,6 +109,31 @@ public sealed class AudioTranscriptionRequest : IDisposable
...
 
+        public AudioTranscriptionRequest(
+            Stream audioStream,
+            string model = null,
+            string prompt = null,
+            AudioResponseFormat responseFormat = AudioResponseFormat.Json,
+            float? temperature = null,
+            string language = null,
+            TimestampGranularity timestampGranularity = TimestampGranularity.None)
+            : this(audioStream, $"UserSpeech.wav", model, prompt, responseFormat, temperature, language, timestampGranularity)
```
Comment (StephenHodgson) [inline OpenAI/Packages/com.openai.unity/Runtime/Audio/AudioTranscriptionRequest.cs:120]: make file name required param

File: OpenAI/Packages/com.openai.unity/Runtime/OpenAIClient.cs [hunk 3]
```diff
@@ -64,7 +64,7 @@ protected override void SetupDefaultRequestHeaders()
 #if !UNITY_WEBGL
                 { "User-Agent", "com.openai.unity" },
 #endif
-                { "OpenAI-Beta", "assistants=v1"}
+                { "OpenAI-Beta", "assistants=v2"}
```
Comment (StephenHodgson) [inline OpenAI/Packages/com.openai.unity/Runtime/OpenAIClient.cs:67]: these changes should be broken out into a separate PR from the projectId changes

File: OpenAI/Packages/com.openai.unity/Runtime/OpenAIClient.cs [hunk 4]
```diff
@@ -110,7 +110,7 @@ protected override void ValidateAuthentication()
         /// <summary>
         /// The <see cref="JsonSerializationOptions"/> to use when making calls to the API.
         /// </summary>
-        internal static JsonSerializerSettings JsonSerializationOptions { get; } = new()
+        public static JsonSerializerSettings JsonSerializationOptions { get; } = new()
```
Comment (StephenHodgson) [inline OpenAI/Packages/com.openai.unity/Runtime/OpenAIClient.cs:113]: revert

**Review by StephenHodgson — CHANGES_REQUESTED** [review]
Most of these changes are breaking and will be done in 8.x which needs to first be completed upstream in [OpenAI-DotNet](https://github.com/RageAgainstThePixel/OpenAI-DotNet)

**Comment by ElisMia** [comment]
> Most of these changes are breaking and will be done in 8.x which needs to first be completed upstream in [OpenAI-DotNet](https://github.com/RageAgainstThePixel/OpenAI-DotNet)

That would be awesome.  Do you know when version 8 will be available?

**Comment by StephenHodgson** [comment]
> That would be awesome. Do you know when version 8 will be available?

Hard to say as I do this in my free time.

**Comment by Bodekaer** [comment]
@dnnkeeper would this temporary PR add support for Assistants v2 as well? 
Info: https://platform.openai.com/docs/assistants/migration

**Comment by StephenHodgson** [comment]
Closing this in favor of #235

**Pull request closed without merging**